lib/extendOrdWithCustom.js

```diff
@@ -0,0 +1,68 @@
+const path = require("path");
+const cds = require("@sap/cds");
+const _ = require('lodash');
+
+const CONTENT_MERGE_KEY = 'ordId'
+
+
+function cleanNullProperties(obj) {
+    for (const key in obj) {
+        if (obj[key] === null) {
+            delete obj[key];
+        } else if (typeof obj[key] === 'object') {
+            cleanNullProperties(obj[key]);
+        }
+    }
+    return obj;
+}
+
+function patchGeneratedOrdResources(destinationObj, sourceObj) {
+    const destObj = _.keyBy(destinationObj, CONTENT_MERGE_KEY);
+    const srcObj = _.keyBy(sourceObj, CONTENT_MERGE_KEY);
+    for (const ordId in srcObj) {
+        if (ordId in destObj) {
+            destObj[ordId] = _.assignWith(structuredClone(destObj[ordId]), structuredClone(srcObj[ordId]));
+        } else {
+            destObj[ordId] = srcObj[ordId];
+        }
+    }
+    return cleanNullProperties(Object.values(destObj));
+}
+
+function compareAndHandleCustomORDContentWithExistingContent(ordContent, customORDContent, logger) {
+    const clonedOrdContent = structuredClone(ordContent);
+    for (const key in customORDContent) {
+        const propertyType = typeof customORDContent[key];
+        if (propertyType !== 'object' && propertyType !== 'array') {
```

Comment:
@aramovic79 , these are good comments. 
One of the use cases for this overwrite as well as also the @ORD.Extension is to allow a developer being ahead of the plugins Support for ORD. If we restrict the properties, we can't add things not yet available in the plug-in. But you are right, this comes with the risk of an broken ORD document. We can mitigate the risk with the validator and in case we want to experiment with new ORD features, we might even want the chance to do that.

Reply:
Thanks, I link this comment to the https://github.com/cap-js/ord/issues/65, lets fix it in the following pr.